samples/topology/src/main/java/quarks/samples/topology/JobPollingSample.java

```diff
@@ -0,0 +1,130 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+package quarks.samples.topology;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import quarks.execution.Job;
+import quarks.execution.JobRegistryService;
+import quarks.providers.development.DevelopmentProvider;
+import quarks.runtime.etiao.JobRegistry;
+import quarks.topology.TStream;
+import quarks.topology.Topology;
+
+/**
+ * Job monitoring by polling job state. 
+ * <p>
+ * Demonstrates job monitoring using the {@link JobRegistryService} service.
+ * The example starts a system monitoring application then concurrently 
+ * submits two jobs.  The monitoring application is using a polling source
+ * to periodically scan the job registry and generates tuples containing the 
+ * current state of registered jobs. Tuples are pushed to a sink which prints
+ * them onto the system output.
+ */
+public class JobPollingSample {
+    private final DevelopmentProvider dtp;
+    
+    public static void main(String[] args) throws Exception {
+        
+        JobPollingSample app = new JobPollingSample();
+
+        // Start monitoring app
+        app.monitor();
+
+        Thread.sleep(2000);
+
+        // Asynchronously start two jobs
```

Comment:
Right, I intend to replace this sample with a system application which uses an events source rather than polling to demonstrate the correct way for monitoring jobs.  Any inline sleeps a inserted to simulate a real case scenario, I'll properly document them.